uml/__init__.py

```diff
@@ -349,6 +350,41 @@ def activity_final(self):
 Activity.final = activity_final  # Add to class via monkey patch
 
 
+def activity_input_value(self, name: str, param_type: str, optional: bool = False,
+                       default_value: ValueSpecification = None) -> ActivityParameterNode:
+    """Add an input, then return the ActivityParameterNode that refers to that input
+
+    :param self: Activity
+    :param name: Name of the input
+    :param param_type: type of value expected for the input
+    :param optional: True if the Parameter is optional; default is False
+    :param default_value: if the input is optional, a default value must be set
+    :return: ActivityParameterNode associated with the input
+    """
+    parameter = self.add_input(name=name, param_type=param_type, optional=optional, default_value=default_value)
+    node = ActivityParameterNode(parameter=parameter.property_value)
+    self.nodes.append(node)
+    return node
+Activity.input_value = activity_input_value  # Add to class via monkey patch
+
+
+def activity_designate_output(self, name: str, param_type: str, source: ActivityNode) -> ActivityParameterNode:
+    """Add an output, connect it to an ActivityParameterNode, and get its value from the designated node
+
+    :param self: Activity
+    :param name: Name of the output
+    :param param_type: type of value expected for the output
+    :param source: ActivityNode whose ObjectValue output should be routed to the source
+    :return: ActivityParameterNode associated with the output
+    """
+    parameter = self.add_output(name=name, param_type=param_type)
+    node = ActivityParameterNode(parameter=parameter.property_value)
+    self.nodes.append(node)
+    self.use_value(source, node)
+    return node
+Activity.designate_output = activity_designate_output  # Add to class via monkey patch
```

Comment:
#43 addressed by this.

Reply:
@jakebeal I don't see where the ActivityParameterNodes are defined?  They aren't in your mods to the uml.ttl.

Reply:
Nevermind.  I see it defined in the branch, but for some reason this review has only one commit's changes (?).

Reply:
It might have given you an odd view of it since I merged your own recently merged material in...